### PR TITLE
Add Custom Recipe Category support to KubeJS

### PR DESCRIPTION
**Pull request by krossgg**
## What
This PR adds a GTRecipeCategory builder which can be used by pack developers in KubeJS. It also cleans up various elements relating to Recipe Categories.

### CODE REVIEW

File: src/main/java/com/gregtechceu/gtceu/common/data/GTRecipeTypes.java [hunk 1]
```diff
@@ -98,8 +96,7 @@ public class GTRecipeTypes {
             .setSlotOverlay(false, false, GuiTextures.FURNACE_OVERLAY_1)
             .setProgressBar(GuiTextures.PROGRESS_BAR_ARROW, LEFT_TO_RIGHT)
             .setSteamProgressBar(GuiTextures.PROGRESS_BAR_ARROW_STEAM, LEFT_TO_RIGHT)
-            .setSound(GTSoundEntries.FURNACE)
-            .setXEIVisible(false);
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/common/data/GTRecipeTypes.java:102]: why add these to recipe viewers?

File: src/main/java/com/gregtechceu/gtceu/integration/emi/recipe/GTRecipeEMICategory.java [hunk 2]
```diff
@@ -70,6 +81,38 @@ public static void registerWorkStations(EmiRegistry registry) {
...
+                        .map(CATEGORIES)
+                        .collect(Collectors.toMap(c -> c, man::getRecipes));
+                SET_PAGES_MH.invokeExact(m, (EmiIngredient) EmiStack.EMPTY);
+            } catch (Throwable e) {
+                GTCEu.LOGGER.info("Recipe display issue {}", e.getMessage());
+                EmiApi.displayRecipeCategory(CATEGORIES.apply(type.getCategory()));
+            }
+        } else {
+            EmiApi.displayRecipeCategory(CATEGORIES.apply(type.getCategory()));
+        }
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/integration/emi/recipe/GTRecipeEMICategory.java]: hear me out: ask EMI for api on this?? this is so wrong

**Review by screret — CHANGES_REQUESTED** [review]

File: src/main/java/com/gregtechceu/gtceu/common/data/GTRecipeTypes.java (hunk 1 above)
Comment (krossgg) [reply src/main/java/com/gregtechceu/gtceu/common/data/GTRecipeTypes.java:102]: XEI invisible = nothing gets registered to XEI, not even workstations
I have removed the furnace recipe category from the `RECIPE_CATEGORIES` registry, so the electric furnace recipes will not be registered to recipe viewers. Also have added a special case in each of the `GTXEICategory` classes.

Comment (krossgg) [reply src/main/java/com/gregtechceu/gtceu/common/data/GTRecipeTypes.java:102]: This lets us remove the hardcoded furnace workstation registration.

**Review by screret — APPROVED** [review]

**Pull request merged**